crypto/txscript/src/data_stack.rs

```diff
@@ -90,47 +179,26 @@ impl OpcodeData<i64> for Vec<u8> {
     }
 
     #[inline]
-    fn serialize(from: &i64) -> Self {
-        let sign = from.signum();
-        let mut positive = from.abs();
-        let mut last_saturated = false;
-        let mut number_vec: Vec<u8> = iter::from_fn(move || {
-            if positive == 0 {
-                if last_saturated {
-                    last_saturated = false;
-                    Some(0)
-                } else {
-                    None
-                }
-            } else {
-                let value = positive & 0xff;
-                last_saturated = (value & 0x80) != 0;
-                positive >>= 8;
-                Some(value as u8)
-            }
-        })
-        .collect();
-        if sign == -1 {
-            match number_vec.last_mut() {
-                Some(num) => *num |= 0x80,
-                _ => unreachable!(),
-            }
+    fn serialize(from: &i64) -> Result<Self, SerializationError> {
+        if from == &i64::MIN {
+            return Err(SerializationError::NumberTooLong(*from));
         }
-        number_vec
+        Ok(serialize_i64(from))
     }
 }
 
 impl OpcodeData<i32> for Vec<u8> {
     #[inline]
     fn deserialize(&self) -> Result<i32, TxScriptError> {
         let res = OpcodeData::<i64>::deserialize(self)?;
-        i32::try_from(res.clamp(i32::MIN as i64, i32::MAX as i64))
-            .map_err(|e| TxScriptError::InvalidState(format!("data is too big for `i32`: {e}")))
+        // TODO: Consider getting rid of clamp, since the call to deserialize should return an error
```

Comment:
I don't understand this TODO. The call to desrialize uses i64 deserialization. So this clamp is definitely needed and removing it will be a consensus change

Reply:
cc @someone235 
There are only 2 operations that work with i32. Roll and pick 
Max stack size is a const equal to 244.

We have never applied this clamp, that's why it's not a breaking change.